firebolt_provider/hooks/firebolt.py

```diff
@@ -111,11 +111,21 @@ def get_ui_field_behaviour() -> Dict:
             },
         }
 
-    def __init__(self, *args: Optional[str], **kwargs: Optional[str]) -> None:
+    def __init__(
+        self,
+        database: Optional[str] = None,
+        engine_name: Optional[str] = None,
+        query_timeout: Optional[float] = None,
+        fail_on_query_timeout: bool = True,
+        *args: Optional[str],
+        **kwargs: Optional[str],
+    ) -> None:
         """Firebolthook Constructor"""
         super().__init__(*args, **kwargs)
-        self.database = kwargs.pop("database", None)
-        self.engine_name = kwargs.pop("engine_name", None)
+        self.database = database
+        self.engine_name = engine_name
```

Comment:
Not sure if this matters, but before we would pass parameters like `database` down to the parent class. Now that we've split those from kwargs we no longer do so.
Let's check this does not affect anything else behind the scenes.

Reply:
Just double checked that parent class doesn't take database and engine parameters into account, we should be safe to not propagate them